plugins/secom_cl.py

```diff
@@ -196,322 +221,463 @@ def sem_roi_to_ccd(escan, ccd, roi, margin=0):
-        self.ccd.data.unsubscribe(self._receive_ccd_data)
-        self.ccd.data.synchronizedOn(None)
+        cpos = self._get_center_pos(data, self.ccd_roi)
+        sname = self._streams[n].name.value
 
-    def _receive_ccd_data(self, df, data):
-        """
-        Store CCD data
-        """
-        self._ccd_data.append(data)
-        self._ccd_data_received.set()
+        data.metadata[model.MD_DESCRIPTION] = sname
+        # update center position of optical image (should be the same for all optical images)
+        data.metadata[model.MD_POS] = cpos
 
-    def acquire_ccd(self, x, y, ccd_roi_idx):
-        """
-        Acquire an image from the CCD while having the e-beam at a spot position
-        x, y (floats): spot position in the ebeam coordinates
-        ccd_roi_idx: slice to crop the CCD image
-        return (model.DataArray of shape (Y,X), model.DataArray of shape (1,1)):
-          the CCD image and the SEM data (at the spot)
-        """
-        self.move_spot(x, y)
+        fn = self.filename.value
+        logging.debug("Will save CL data to %s", fn)
+        fn_prefix, fn_ext = os.path.splitext(self.filename.value)
 
-        # Start next CCD acquisition
-        self._ccd_data = []
-        self._ccd_data_received.clear()
-        self.ccd.softwareTrigger.notify()
+        self.save_data(data,
+                       prefix=fn_prefix,
+                       xres=self.repetition.value[0],
+                       yres=self.repetition.value[1],
+                       xstepsize=self._getPixelSize()[0] * 1e9,
+                       ystepsize=self._getPixelSize()[1] * 1e9,
+                       xpos=i[1]+1,  # start counting with 1
+                       ypos=i[0]+1
+                       )
 
-        # Wait for the CCD
-        expt = self.ccd.exposureTime.value
-        if not self._ccd_data_received.wait(expt * 2 + 2):
-            raise IOError("Timed out: No CCD data received in time")
+        # return data
+        return model.DataArray(numpy.array([0]))  # return something
 
-        if len(self._ccd_data) != 1:
-            logging.warning("Received %d CCD data, while expected 1", len(self._ccd_data))
-        d = self._ccd_data[0]
-        d = d[ccd_roi_idx]  # crop
+    def _onCompletedData(self, n, raw_das):
+        """
+        TODO
+        :param n: (int) The index of the detector.
+        :param raw_das: (list) List of data acquired for given detector n.
+        """
+        if n != self._ccd_idx:
+            return super(SECOMCLSEMMDStream, self)._onCompletedData(n, raw_das)
 
-        if not self._sem_data_received.wait(3):
-            logging.warning("No SEM data received, 3s after the CCD data")
-        if len(self._sem_data) > 1:
-            logging.warning("Received %d SEM data, while expected just 1", len(self._sem_data))
+        memu = acqstream.udriver.readMemoryUsage()
+        # Too bad, need to use VmSize to get any good value
+        logging.info("Memory usage is %d. ", memu)
```

Comment:
This memory check can go away now that we've confirmed the memory doesn't go crazy.